File: code/functions/pyprism/pyprism/data/_load_cln.py
# -*- coding: utf-8 -*-
"""
@created: Apr 22 2021
@created: Oct 01 2020
@author: Yoann Pradat

    CentraleSupelec
    MICS laboratory
    9 rue Juliot Curie, Gif-Sur-Yvette, 91190 France

    Institut Gustave Roussy
    Prism Center
    114 rue Edouard Vaillant, Villejuif, 94800 France

Code for the user function load_cln.
"""

import os
import numpy as np
import pandas as pd
import re
import sys

from typing import Iterable
from ._util_load import load_from_data, subset_data
from ._filepaths import _get_filepath_cln_curated

DataFrame = pd.core.frame.DataFrame

# User function ========================================================================================================

def load_cln(study: str, identifiers: Iterable[str]=None, identifiers_name: str=None, mode:str="in_design", **kwargs) -> DataFrame:
    """
    Load clinical tables for the specified study.

    Parameters
    ----------
    study: str
        Choose one of 'tcga' or 'prism'.
    identifiers: Iterable[str], optional.
        If not None, return annotations only for the identifiers values.
    identifiers_name: Iterable[str], optional.
        Name of the column of identifiers.
    mode: str, optional
        Either 'in_design' or 'all' for 'pancancer' annotations or 'brca' for per-cancer annotations.

    Returns
    -------
    df_cln: DataFrame
    """

    df_cln = load_from_data(_get_filepath_cln_curated(study=study, mode=mode), **kwargs)
    df_cln = subset_data(df_cln, values=identifiers, col_name=identifiers_name)

    return df_cln

File: code/functions/rprism/R/load_cln.R
# @created: 25 Jan 21
# @modified: 27 Apr 21
# @author: Yoann Pradat
# 
# Exported functions
#     load_cln

#' Load clinical files.
#'
#' This function loads clinical tables (one line is one subject id) for the specified study and optionally specified
#' identifiers. The \code{mode} parameters allow to choose which table should be loaded between 'in_design' and 'all'.
#'
#' @return a data.frame
#'
#' @param study Name of the cohort used in the naming of the files. Choose "met500", "prism" or "tcga".
#' @param identifiers (optional) If not NULL, return clinical tables only for the individuals specified.
#' @param identifiers_name (optional) The name of the column the \code{identifiers} values correspond to.
#' @param mode (optional) Either 'in_design' or 'all' for 'pancancer' annotations or 'brca' for per-cancer annotations.
#' @param ... Extra parameters passed to \code{\link{load_from_data}}
#'
#' @author Yoann Pradat
#'
#' @export
load_cln <- function(study, identifiers=NULL, identifiers_name=NULL, mode="in_design", ...){
  df_cln <- load_from_data(get_filepath_cln_curated(study, mode), progress=F, show_col_types=F, ...)
  df_cln <- subset_data(df=df_cln, values=identifiers, col_name=identifiers_name)

  return(df_cln)
}
